stack/new_stack.py

```diff
@@ -0,0 +1,176 @@
+"""Create GCP project + stack file for Pulumi"""
+import os
+import re
+from subprocess import check_output
+
+import yaml
+import click
+
+ORGANIZATION_ID = 648561325637
+BILLING_ACCOUNT_ID = '01D012-20A6A2-CBD343'
+BILLING_PROJECT_ID = 'billing-admin-290403'
+HAIL_PROJECT = 'hail-295901'
+
+RE_CLIENT_EMAIL_MATCHER = re.compile(
+    f'[A-z0-9-]+@{HAIL_PROJECT}.iam.gserviceaccount.com'
+)
+
+GET_HAIL_TOKENS = (
+    lambda project: f"""
+for access_level in test standard full; do kubectl get secret {project}-$access_level-gsa-key -o json | jq '.data | map_values(@base64d)'; done
+"""
+)
+
+
+@click.command()
+@click.option('--dataset')
+@click.option('--gcp-project', required=False, help='If different to the dataset name')
+@click.option('--release', required=False, is_flag=True)
+def main(
+    dataset: str,
+    gcp_project: str = None,
+    release=False,
+):
+    """Function that coordinates creating a project"""
+    _gcp_project = gcp_project or dataset
+
+    if len(dataset) > 17:
+        raise ValueError(
+            f'The dataset length must be less than 17 characters (got {len(dataset)})'
+        )
+
+    pulumi_config_fn = f'Pulumi.{dataset}.yaml'
+    if os.path.exists(pulumi_config_fn):
+        raise ValueError('The pulumi config already exists')
+
+    inp = str(
+        input('Please confirm you have created the accounts in Hail Batch (y/n): ')
+    )
+    if inp not in ('y', '1', 't'):
+        raise SystemExit
+
+    create_project(project_id=_gcp_project)
+    assign_billing_account(_gcp_project)
+    create_budget(_gcp_project)
+    hail_client_emails_by_level = get_hail_service_accounts(dataset=dataset)
+
+    formed_hail_config = {
+        f'datasets:hail_service_account_{access_level}': value
+        for access_level, value in hail_client_emails_by_level.items()
+    }
+    pulumi_stack = {
+        'encryptionsalt': 'v1:SPQXlVggjxw=:v1:X1sTpViNuyK+Wcom:3GANI8gZOKtk0gG7BklsyHeNU5uVLw==',
+        'config': {
+            'datasets:archive_age': '90',
+            'datasets:customer_id': 'C010ys3gt',
+            'datasets:enable_release': release,
+            'gcp:billing_project': _gcp_project,
+            'gcp:project': dataset,
+            'gcp:user_project_override': 'true',
+            **formed_hail_config,
+        },
+    }
+
+    with open(pulumi_config_fn, 'w+', encoding='utf-8') as fp:
+        print(f'Writing to {pulumi_config_fn}')
+        yaml.dump(pulumi_stack, fp, default_flow_style=False)
+
+    command = f"""\
+    export PULUMI_CONFIG_PASSPHRASE=
+    pulumi stack init {dataset}
+    pulumi stack select {dataset}
+    pulumi up
```

Comment:
Is it worth adding `git` + `gh` commands here as well, to make sure the new config gets merged?

Reply:
I've parameterised this a lot, and it commits automatically (kind of like bumpversion) when you run the step that creates the config.